### PR TITLE
Use Prism v0.27.0

### PR DESCRIPTION
**Pull request by andyw8**

### CODE REVIEW

File: lib/ruby_lsp/internal.rb [hunk 1]
```diff
@@ -19,14 +19,6 @@
 require "prism/visitor"
 require "language_server-protocol"
 
-# Prism v0.26.0 introduced generics for ParseResult, but it causes some problems so the intention is to remove it.
-# Once that is done, we can remove this patch.
-module Prism
-  class ParseResult
-    extend T::Generic
-  end
-end
```
Comment (andyw8) [inline lib/ruby_lsp/internal.rb:28]: Actually we will need to conditionally include this... will think how to handle that.

File: ruby-lsp.gemspec [hunk 2]
```diff
@@ -19,7 +19,7 @@ Gem::Specification.new do |s|
   s.require_paths = ["lib"]
 
   s.add_dependency("language_server-protocol", "~> 3.17.0")
-  s.add_dependency("prism", ">= 0.23.0", "< 0.27")
+  s.add_dependency("prism", ">= 0.23.0", "< 0.28")
```
Comment (andyw8) [inline ruby-lsp.gemspec:22]: @vinistock should we bump the minimum to `>= 0.27.0` now?

File: lib/ruby_lsp/listeners/document_link.rb [hunk 3]
```diff
@@ -124,7 +124,7 @@ def extract_document_link(node)
         match = comment.location.slice.match(%r{source://.*#\d+$})
         return unless match
 
-        uri = T.cast(URI(match[0]), URI::Source)
+        uri = T.cast(URI(T.must(match[0])), URI::Source)
```
Comment (st0012) [inline lib/ruby_lsp/listeners/document_link.rb:127]: Could this `T.must` cause problem when used with Prism `0.26`?

**Review by st0012 — APPROVED** [review]
Just have one question, otherwise all good 👍

File: lib/ruby_lsp/listeners/document_link.rb (hunk 3 above)
Comment (vinistock) [reply lib/ruby_lsp/listeners/document_link.rb:127]: I don't believe there are runtime errors for calling `T.must` on an untyped value. It's just a static check, which shouldn't impact users of the LSP.

**Pull request merged**